src/main/scala/zio/prelude/fx/ZPure.scala

```diff
@@ -35,13 +36,13 @@ sealed trait ZPure[-S1, +S2, -R, +E, +A] { self =>
     self zipRight that
 
   /**
-   * Runs this computatation if the provided environment is a `Left` or else
+   * Runs this computation if the provided environment is a `Left` or else
    * runs that computation if the provided environment is a `Right`, returning
    * the result in an `Either`.
    */
   final def +++[S0 <: S1, S3 >: S2, R1, B, E1 >: E](
     that: ZPure[S0, S3, R1, E1, B]
-  ): ZPure[S0, S3, Either[R, R1], E1, Either[A, B]] =
+  )(implicit evL: NeedsEnv[R], evR: NeedsEnv[R1]): ZPure[S0, S3, Either[R, R1], E1, Either[A, B]] =
```

Comment:
Don't think this is correct. Even if the environment type is any this is well defined. Just if we provide `Left` with anything we get one value and if we provide `Right` with anything we get the other value.